cmd/alertmanager/main.go

```diff
@@ -382,31 +378,106 @@ func main() {
 	level.Info(logger).Log("msg", "Received SIGTERM, exiting gracefully...")
 }
 
+type meshCollector struct {
+	router   *mesh.Router
+	connDesc *prometheus.Desc
+	posDesc  *prometheus.Desc
+	termDesc *prometheus.Desc
+}
+
+func NewMeshCollector(router *mesh.Router) *meshCollector {
+	return &meshCollector{
+		router: router,
+		connDesc: prometheus.NewDesc(
+			"alertmanager_peer_connection",
+			"State of the connection between the Alertmanager instance and a peer.",
+			[]string{"peer", "nick"},
```

Comment:
Are both of these labels needed?

Reply:
My reasoning is that the nickname makes it easier to identify the peer:

```
alertmanager_peer_connection{instance="prometheus-0",job="alertmanager",nick="prometheus-1",peer="02:42:ac:11:00:03"}  1
alertmanager_peer_connection{instance="prometheus-1",job="alertmanager",nick="prometheus-0",peer="02:42:ac:11:00:02"}  1
```

But we can't get rid of of `peer` altogether since `nick`'s uniqueness isn't enforced by the mesh library.

Reply:
these are both belonging to the connected AM, I imagine only one is necessary

Reply:
I would vote for only using `peer`. If a user wants to make it easier to identify they can set a custom peer-id.

Reply:
Fine with me. I'll update the PR accordingly.